resources/js/mp-history/score.tsx

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import FlagCountry from 'components/flag-country';
+import Mod from 'components/mod';
+import Ruleset from 'interfaces/ruleset';
+import ScoreJson, { ScoreStatisticsAttribute } from 'interfaces/score-json';
+import UserJson from 'interfaces/user-json';
+import { route } from 'laroute';
+import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
+import { formatNumber } from 'utils/html';
+import { trans } from 'utils/lang';
+
+interface Props {
+  mode: Ruleset;
+  score: ScoreJson;
+  users: Partial<Record<number, UserJson>>;
+}
+
+const firstRow = ['combo', 'accuracy', 'score'];
+const secondRow: ScoreStatisticsAttribute[] = ['count_geki', 'count_300', 'count_katu', 'count_100', 'count_50', 'count_miss'];
+
+export default function Score(props: Props) {
+  const user = props.users[props.score.user_id];
+
+  if (user == null) {
+    throw new Error('user for score is missing');
+  }
+
+  if (props.score.match == null) {
+    throw new Error('score is missing match data');
+  }
+
+  return (
+    <div className='mp-history-game__player-score mp-history-player-score'>
+      <div
+        className='mp-history-player-score__shapes'
+        style={{ backgroundImage: `url(/images/layout/mp-history/shapes-team-${props.score.match.team ?? 'none'}.svg)` }} />
+      <div className='mp-history-player-score__main'>
+        <div className={classWithModifiers('mp-history-player-score__info-box', ['user'])}>
+          <div className='mp-history-player-score__username-box'>
+            <a className='mp-history-player-score__username' href={route('users.show', { user: user.id })}>{user.username}</a>
+            {!props.score.match.pass
+              ? <span className='mp-history-player-score__failed'>{trans('matches.match.failed')}</span>
+              : null}
```

Comment:
`false` is [special cased](https://react.dev/learn/conditional-rendering#logical-and-operator-) in react so if there's nothing to render when false, `{testCondition && <Thing />}` works (apparently also for `true`/`||` but doesn't seem to be documented?)